rskj-core/src/main/java/co/rsk/util/HexUtils.java

```diff
@@ -0,0 +1,336 @@
+        return input.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Converts a byte array to a string according to ethereum json-rpc specifications
+     */
+    public static String toJsonHex(final byte[] param) {
+        String result = toUnformattedJsonHex(param);
+
+        if (HEX_PREFIX.equals(result)) {
+            return "0x00";
+        }
+
+        return result;
+    }
+
+    /**
+     * Convert from Coin to its BigInteger value in String representation or empty if the parameter is null
+     */
+    public static String toJsonHex(final Coin coin) {
+        return coin != null ? coin.asBigInteger().toString() : "" ;
+    }
+
+    /**
+     * add 0x prefix
+     */
+    public static String toJsonHex(final String param) {
+        return HEX_PREFIX + param;
+    }
+
+    /**
+     * @return A Hex representation of n WITHOUT leading zeroes
+     */
+    public static String toQuantityJsonHex(long n) {
+        return HEX_PREFIX + Long.toHexString(n);
+    }
+
+    /**
+     * @return A Hex representation of n WITHOUT leading zeroes
+     */
+    public static String toQuantityJsonHex(BigInteger n) {
+        return HEX_PREFIX + n.toString(16);
+    }
+
+    /**
+     * Converts a byte array to a string according to ethereum json-rpc specifications, null and empty
+     * convert to 0x.
+     *
+     * @param param An unformatted byte array
+     * @return A hex representation of the input with two hex digits per byte
+     */
+    public static String toUnformattedJsonHex(byte[] param) {
+        return HEX_PREFIX + (param == null ? "" : ByteUtil.toHexString(param));
+    }
+
+    /**
+     * Converts a byte array representing a quantity according to ethereum json-rpc specifications.
+     *
+     * <p>
+     * 0x000AEF -> 0x2AEF
+     * <p>
+     * 0x00 -> 0x0
+     * @param x A hex string with or without leading zeroes ("0x00AEF"). If null, it is considered as zero.
+     * @return A hex string without leading zeroes ("0xAEF")
+     */
+    public static String toQuantityJsonHex(byte[] x) {
+        String withoutLeading = LEADING_ZEROS_PATTERN.matcher(toJsonHex(x)).replaceFirst("0x");
+        if (HEX_PREFIX.equals(withoutLeading)) {
+            return "0x0";
+        }
+
+        return withoutLeading;
+    }
+
+    /**
+     * decodes a hexadecimal encoded with the 0x prefix into a long
+     */
+    public static long jsonHexToLong(String x) {
+        if (!hasHexPrefix(x)) {
+            throw new NumberFormatException(INCORRECT_HEX_SYNTAX);
+        }
+        x = removeHexPrefix(x);
+        return Long.parseLong(x, 16);
+    }
+
+    /**
+     * if the parameter has the hex prefix 
+     */
+    public static boolean hasHexPrefix(final String data) {
+        return data != null && data.startsWith(HEX_PREFIX);
+    }
+
+    /**
+     * if the parameter has the hex prefix 
+     */
+    public static boolean hasHexPrefix(final byte[] data) {
+
+        if (data == null) {
+            return false;
+        }
+
```

Comment:
should we check `data` length here instead of checking it in the caller?

Reply:
yes